src/pages/Main/MainCarousel/MainCarousel.js

```diff
@@ -0,0 +1,144 @@
+  },
+  {
+    eventId: 4,
+    mainCategoryIdLink: 4,
+    title: "세대별 인기 클래스 선착 할인!",
+    subTitle: "내 주변에서 가장 인기있는 클래스는 무엇?",
+    url: "https://s3.us-west-2.amazonaws.com/secure.notion-static.com/b925371d-ea09-4a03-8039-8fb04ea96cc0/Untitled.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20220511%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220511T085309Z&X-Amz-Expires=86400&X-Amz-Signature=758120dd2ae9ca76a23d90d97c2c71662108c2c1a79787a450cf97c738d81278&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Untitled.png%22&x-id=GetObject",
+  },
+];
+
+function MainCarousel() {
+  return (
+    <CarouselWrapper>
+      <Main>
+        <Carousel {...settings}>
+          {MOCK.map((caruselData, i) => (
+            <MainCarouselContainer
+              backColor={caruselData.backColor}
+              url={caruselData.url}
+              key={i}
+            >
+              <BackImg src={caruselData.url} />
+              <MainCarouselBox>
+                <MainCarouselWrapper to="/mainCategory">
+                  <CarouselImg src={caruselData.url} />
+                </MainCarouselWrapper>
+                <div>
+                  <CarouselText to="/products/main_category" main>
+                    {caruselData.title}
+                  </CarouselText>
+                  <CarouselText to="/products/main_category">
+                    {caruselData.subTitle}
+                  </CarouselText>
+                </div>
+              </MainCarouselBox>
+            </MainCarouselContainer>
+          ))}
+        </Carousel>
+      </Main>
+    </CarouselWrapper>
+  );
+}
+
+export default MainCarousel;
+
+const settings = {
+  infinite: true,
+  speed: 300,
+  slidesToShow: 1,
+  slidesToScroll: 1,
+  autoplay: true,
+  autoplaySpeed: 4000,
+  arrows: false,
+  cssEase: "Ease-out",
+  // fade: true,
+};
+
+const CarouselWrapper = styled.div`
+  display: flex;
+  position: relative;
+  justify-content: center;
+  width: 100%;
+  margin-top: 105px;
+`;
+
+const Main = styled.div`
+  width: 100%;
+  height: 100%;
+`;
+
+const Carousel = styled(Slider)``;
+
+const MainCarouselContainer = styled.div`
+  position: relative;
+  height: 100%;
+`;
+
+const BackImg = styled.img`
+  position: absolute;
+  z-index: -90;
+  width: 100vw;
+  background-size: cover;
+
+  filter: blur(10px) brightness(25%);
+`;
+
+const MainCarouselBox = styled.div`
+  width: 1176px;
+  display: flex;
+  margin: 0 auto;
+`;
+
+const MainCarouselWrapper = styled(Link)`
+  width: 675px;
+  height: 415px;
+`;
+
+const CarouselImg = styled.img`
+  width: 675px;
+  height: 415px;
+  object-fit: cover;
+  border-radius: 10px;
+  padding: 20px;
+`;
+
+const CarouselText = styled.div`
+  margin-top: ${props => (props.main ? "7%" : "10%")};
```

Comment:
```suggestion
  margin-top: ${({main}) => (main ? "7%" : "10%")};
```
* CSS in JS이기 때문에, 자바스크립트 문법대로 구조분해 할당도 됩니다.

Reply:
`  margin-top: ${({ main }) => (main ? "7%" : "10%")};
  margin-left: 10%;
  color: ${({ main }) => (main ? "white" : "whitesmoke")};
  font-size: ${({ main }) => (main ? "36px" : "16px")};
  line-height: 44px;
  font-weight: ${({ main }) => (main ? "bold" : "normal")};
  letter-spacing: -0.02rem;`
  
  
감사합니다,,! 해당부분 모두 구조분해할당 했습니다,,!